### PR TITLE
Fix sql multimedia export

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan http://manage.dimagi.com/default.asp?229063 this should fix up exporting sql forms. best read 🐟 

cc: @biyeun

### CODE REVIEW

File: corehq/apps/reports/tests/test_esaccessors.py [hunk 1]
```diff
@@ -73,7 +74,8 @@ class TestFormESAccessors(BaseESAccessorsTest):
     pillow_class = XFormPillow
     es_index = XFORM_INDEX
 
-    def _send_form_to_es(self, domain=None, completion_time=None, received_on=None, **metadata_kwargs):
+    def _send_form_to_es(self, domain=None, completion_time=None, received_on=None, attachment_dict=None, **metadata_kwargs):
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_esaccessors.py]: E501 line too long (125 > 115 characters)

**Comment by benrudolph** [comment]
i'm tempted to say we don't have to reindex since there's no real data that has been submitted

**Comment by czue** [comment]
(if you want you can also manually run a sql reindex which completes in seconds on prod.)

**Comment by benrudolph** [comment]
good point, will do that

File: corehq/apps/reports/tasks.py [hunk 2]
```diff
@@ -46,6 +44,12 @@
 from corehq.util.files import file_extention_from_filename
 from corehq.util.view_utils import absolute_reverse
 
+from .analytics.couchaccessors import (
+    get_form_ids_having_multimedia as couch_get_form_ids_having_multimedia
+)
+from .analytics.esaccessors import (
+    get_form_ids_having_multimedia as es_get_form_ids_having_multimedia
```
Comment (snopoke) [inline corehq/apps/reports/tasks.py]: Could just name the functions differently which would also prevent anyone importing the wrong one by accident.

File: corehq/apps/reports/tasks.py [hunk 3]
```diff
@@ -381,49 +385,24 @@ def _get_export_properties(export_id, export_is_legacy):
...
-            f['id'] for f in
-            XFormInstance.get_db().view(
-                "attachments/attachments",
-                start_key=key + [startdate],
-                end_key=key + [enddate, {}],
-                reduce=False
-            )
-        }
-    return form_ids
+        fetch_fn = couch_get_form_ids_having_multimedia
```
Comment (snopoke) [inline corehq/apps/reports/tasks.py]: nit but I'd swap the if logic around - always easier to read if you don't have to boolean logic in your head.

**Comment by snopoke** [comment]
looks good. Build failure real:
```
======================================================================
ERROR: test_basic_completed_by_user (corehq.apps.reports.tests.test_esaccessors.TestFormESAccessorsSQL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_esaccessors.py", line 238, in test_basic_completed_by_user
    self._send_form_to_es(completion_time=datetime(2013, 7, 2))
  File "/mnt/commcare-hq-ro/corehq/apps/reports/tests/test_esaccessors.py", line 94, in _send_form_to_es
    form_pair.wrapped_form._attachments.update(attachment_dict)
AttributeError: 'XFormInstanceSQL' object has no attribute '_attachments'
```

**Comment by nickpell** [comment]
The test failure here is real and is breaking the build

**Pull request merged**